internal/operator-controller/applier/helm_test.go

```diff
@@ -386,3 +443,76 @@ func TestApply_Upgrade(t *testing.T) {
 		assert.Equal(t, "service-b", objs[1].GetName())
 	})
 }
+
+func TestApply_InstallationWithSingleOwnNamespaceInstallSupportEnabled(t *testing.T) {
+	featuregatetesting.SetFeatureGateDuringTest(t, features.OperatorControllerFeatureGate, features.SingleOwnNamespaceInstallSupport, true)
+
+	t.Run("generates bundle resources in SingleNamespace install mode when watch namespace is configured", func(t *testing.T) {
+		var expectedWatchNamespace = "watch-namespace"
+
+		helmApplier := applier.Helm{
+			ActionClientGetter: &mockActionGetter{
+				getClientErr: driver.ErrReleaseNotFound,
+				desiredRel: &release.Release{
+					Info:     &release.Info{Status: release.StatusDeployed},
+					Manifest: validManifest,
+				},
+			},
+			BundleToHelmChartFn: func(rv1 fs.FS, installNamespace string, watchNamespace string) (*chart.Chart, error) {
+				require.Equal(t, expectedWatchNamespace, watchNamespace)
+				return convert.RegistryV1ToHelmChart(rv1, installNamespace, watchNamespace)
+			},
+		}
+
+		testExt := &v1.ClusterExtension{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "testExt",
+				Annotations: map[string]string{
+					applier.AnnotationClusterExtensionWatchNamespace: expectedWatchNamespace,
+				},
+			},
+		}
+
+		_, _, _ = helmApplier.Apply(context.TODO(), validFS, testExt, testObjectLabels, testStorageLabels)
```

Comment:
Can we add here a test to ensure that if it is MultiNamespace, then it fails when the flag is enabled?

Reply:
We kinda handle that [here](https://github.com/operator-framework/operator-controller/pull/1724/files#diff-2939d8b5276ecbb3620b1f3a113be76a4de4d85ba41087abe6cf28db93710f19R50. I've also restricted it at the API level with the [RegistryV1ToHelmChart](https://github.com/operator-framework/operator-controller/pull/1724/files#diff-4e1161250663230583e5c1839b774ef8ee35d18cc9fc5801b68516a12bd3dc5eR42) only accepting a single watchNamespace. There's no way to initiate installation in MultiNamespace mode. 

Additionally, for the applier suite I'd really only want to test integration between the applier and the converter. I.e. is the applier calling the converter correctly and is it floating up errors. For deeper testing I think it's better to focus on the Rukpak package.